test/e2e/set/nodeconfig/nodeconfig_disksetup.go

```diff
@@ -456,6 +457,87 @@ var _ = g.Describe("Node Setup", framework.Serial, func() {
 
 				return cleanupFunc
 			},
+			postNodeConfigCreationFunc: nil,
+		}),
+		g.Entry("for mount unit configured with a corrupted filesystem", &degradedEntry{
+			nodeConfigFunc: func() *scyllav1alpha1.NodeConfig {
+				nc := ncTemplate.DeepCopy()
+				nc.Spec.LocalDiskSetup = &scyllav1alpha1.LocalDiskSetup{
+					LoopDevices: []scyllav1alpha1.LoopDeviceConfiguration{
+						{
+							Name:      "disk",
+							ImagePath: fmt.Sprintf("/mnt/%s.img", f.Namespace()),
+							Size:      resource.MustParse("32M"),
+						},
+					},
+					Filesystems: []scyllav1alpha1.FilesystemConfiguration{
+						{
+							Device: "/dev/loops/disk",
+							Type:   scyllav1alpha1.XFSFilesystem,
+						},
+					},
+				}
+				return nc
+			},
+			preNodeConfigCreationFunc: nil,
+			postNodeConfigCreationFunc: func(ctx context.Context, nc *scyllav1alpha1.NodeConfig) func(context.Context) {
```

Comment:
putting a whole new test into an entry seems weird to me - I think this should be a new `It` block and if there is a common setup then `BeforeEach` seems to address that

Reply:
> BeforeEach seems to address that

iirc you can't nest BeforeEach in container nodes and it's broken otherwise since our framework uses it for initialisation, so you can't enforce any order between multiple BeforeEach blocks

> putting a whole new test into an entry seems weird to me - I think this should be a new It block

there's already a bunch of tests there that are testing propagation of degraded status conditions for a single mount controller, only for different scenarios - so why is that an issue now if it wasn't before? I agree it grows out of proportion a bit but it's either this or a lot of code duplication

Reply:
you are right that it's preexisting